fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/impl/ResourceReferenceDAO.java

```diff
@@ -153,19 +153,46 @@ public CommonTokenValueResult readCommonTokenValueId(String codeSystem, String t
     }
 
     @Override
-    public void addCommonTokenValues(String resourceType, Collection<ResourceTokenValueRec> xrefs) {
-        // Grab the ids for all the code-systems, and upsert any misses
-        List<ResourceTokenValueRec> systemMisses = new ArrayList<>();
-        cache.resolveCodeSystems(xrefs, systemMisses);
-        upsertCodeSystems(systemMisses);
+    public Integer readCanonicalId(String canonicalValue) {
+        Integer result;
+        final String SQL = ""
+                + "SELECT canonical_id "
+                + "  FROM common_canonical_values "
+                + " WHERE url = ? ";
+        try (PreparedStatement ps = connection.prepareStatement(SQL)) {
+            ps.setString(1, canonicalValue);
+            ResultSet rs = ps.executeQuery();
+            if (rs.next()) {
+                result = rs.getInt(1);
+            } else {
+                result = null;
+            }
+        } catch (SQLException x) {
+            logger.log(Level.SEVERE, SQL, x);
+            throw translator.translate(x);
+        }
 
-        // Now that all the code-systems ids are known, we can search the cache
-        // for all the token values, upserting anything new
-        List<ResourceTokenValueRec> valueMisses = new ArrayList<>();
-        cache.resolveTokenValues(xrefs, valueMisses);
-        upsertCommonTokenValues(valueMisses);
+        return result;
+    }
 
-        insertResourceTokenRefs(resourceType, xrefs);
+    @Override
+    public void addNormalizedValues(String resourceType, Collection<ResourceTokenValueRec> xrefs, Collection<ResourceProfileRec> profileRecs, Collection<ResourceTokenValueRec> tagRecs, Collection<ResourceTokenValueRec> securityRecs) {
+        // Grab the ids for all the code-systems, and upsert any misses
+//        List<ResourceTokenValueRec> systemMisses = new ArrayList<>();
```

Comment:
Is this commented out code needed?

Reply:
removed. This was refactored to use a common method.